R/promise.R

```diff
@@ -142,7 +142,9 @@ Promise <- R6::R6Class("Promise",
     then = function(onFulfilled = NULL, onRejected = NULL, onFinally = NULL) {
       onFulfilled <- normalizeOnFulfilled(onFulfilled)
       onRejected <- normalizeOnRejected(onRejected)
-      # TODO: Verify onFinally is NULL or has one argument
+      if (!is.function(onFinally)) {
```

Comment:
This doesn't exactly do what the comment said -- it doesn't check for one argument. Also, is it necessary? If users to pass in a non-function, do we really want to set it to NULL, or would it be better to throw an error?

Reply:
Oh, sorry, I didn't explain this. The Promise R6 object follows the A-Plus spec pretty closely, and that spec is explicit that non-function values for `onFulfilled` and `onRejected` should be accepted without complaint but then they are no-ops. So `onFinally` follows the same guidelines.

OTOH, the functional API we put on top does not follow those guidelines, so `then()`, `catch()`, and `finally()` will all complain if it's not a function, but not complain about the wrong number of parameters. I think that's fine.

The unit tests I added in my last commit verify the above (other than differing number of parameters).